File: src/main/java/org/akanza/model/Partner.java
package org.akanza.model;

import org.hibernate.validator.constraints.Email;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by Christian Amani on 01/05/2017.
 */
@Entity
@Table(name = "partner")
public class Partner extends Receiver
{

    @Column(name = "nom",nullable = false)
    private String name;
    @Column(name = "url_web_site")
    private String urlWebSite;
    @Column(name = "libelle")
    private String words;
    @Column(name = "créer")
    private LocalDate created;
    @Column(name = "numero_1")
    private String numberPhone1;
    @Column(name = "numero_2")
    private String numberPhone2;
    @Column(name = "numero_3")
    private String numberPhone3;
    @Column(name = "mail_1")
    @Email
    private String email1;
    @Column(name = "mail_2")
    @Email
    private String email2;
    @Column(name = "siège")
    private String seat;
    @OneToMany(mappedBy = "partner",cascade = CascadeType.ALL,orphanRemoval = true)
    private List<SmsSent> listSms = new ArrayList<>();

    public Partner()
    {
        // Not Implemented
    }

    public Partner(String name, String urlWebSite, String words, LocalDate created, String numberPhone1,
            String numberPhone2, String numberPhone3, String email1, String email2, String seat)
    {
        this.name = name;
        this.urlWebSite = urlWebSite;
        this.words = words;
        this.created = created;
        this.numberPhone1 = numberPhone1;
        this.numberPhone2 = numberPhone2;
        this.numberPhone3 = numberPhone3;
        this.email1 = email1;
        this.email2 = email2;
        this.seat = seat;
    }

    public String getName()
    {
        return name;
    }

    public void setName(String name)
    {
        this.name = name;
    }

    public String getUrlWebSite()
    {
        return urlWebSite;
    }

    public void setUrlWebSite(String urlWebSite)
    {
        this.urlWebSite = urlWebSite;
    }

    public String getWords()
    {
        return words;
    }

    public void setWords(String words)
    {
        this.words = words;
    }

    public LocalDate getCreated()
    {
        return created;
    }

    public void setCreated(LocalDate created)
    {
        this.created = created;
    }

    public String getNumberPhone1()
    {
        return numberPhone1;
    }

    public void setNumberPhone1(String numberPhone1)
    {
        this.numberPhone1 = numberPhone1;
    }

    public String getNumberPhone2()
    {
        return numberPhone2;
    }

    public void setNumberPhone2(String numberPhone2)
    {
        this.numberPhone2 = numberPhone2;
    }

    public String getNumberPhone3()
    {
        return numberPhone3;
    }

    public void setNumberPhone3(String numberPhone3)
    {
        this.numberPhone3 = numberPhone3;
    }

    public String getEmail1()
    {
        return email1;
    }

    public void setEmail1(String email1)
    {
        this.email1 = email1;
    }

    public String getEmail2()
    {
        return email2;
    }

    public void setEmail2(String email2)
    {
        this.email2 = email2;
    }

    public String getSeat()
    {
        return seat;
    }

    public void setSeat(String seat)
    {
        this.seat = seat;
    }

    public List<SmsSent> getListSms()
    {
        return listSms;
    }
}

File: webui/lib/model/partner.dart
class Partner{
    int _id;
    String _name;
    String _urlWebSite;
    String _words;
    DateTime _created;
    String _numberPhone1;
    String _numberPhone2;
    String _numberPhone3;
    String _email1;
    String _email2;
    String _seat;

    Partner(this._id,this._name,this._urlWebSite,this._words,this._created,this._numberPhone1,this._numberPhone2
    ,this._numberPhone3,this._email1,this._email2,this._seat);

    set id(int id) => _id = id;
    int get id => _id;
    set name(String name) => _name = name;
    String get name => _name;
    String get words => _words;
    set created(DateTime created) => _created = created;
    DateTime get created => _created;
    set urlWebSite(String urlWebSite) => _urlWebSite = urlWebSite;
    String get urlWebSite => _urlWebSite;
    set numberPhone1(String numberPhone1) => _numberPhone1 = numberPhone1;
    set numberPhone2(String numberPhone2) => _numberPhone2 = numberPhone2;
    set numberPhone3(String numberPhone3) => _numberPhone3 = numberPhone3;
    String get numberPhone1 => _numberPhone1;
    String get numberPhone2 => _numberPhone2;
    String get numberPhone3 => _numberPhone3;
    set email1(String email1) => _email1 = email1;
    set email2(String email2) => _email2 = email2;
    String get email1 => _email1;
    String get email2 => _email2;
    set seat(String seat) => _seat = seat;
    String get seat => _seat;
}

/***
 *  @Column(name = "nom",nullable = false)
    private String name;
    @Column(name = "url_web_site")
    private String urlWebSite;
    @Column(name = "libelle")
    private String words;
    @Column(name = "créer")
    private LocalDate created;
    @Column(name = "numero_1")
    private String numberPhone1;
    @Column(name = "numero_2")
    private String numberPhone2;
    @Column(name = "numero_3")
    private String numberPhone3;
    @Column(name = "mail_1")
    @Email
    private String email1;
    @Column(name = "mail_2")
    @Email
    private String email2;
    @Column(name = "siège")
    private String seat;
 */